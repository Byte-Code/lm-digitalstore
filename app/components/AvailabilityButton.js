import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Dialog, FlatButton } from 'material-ui';

import AvailabilityMap from '../containers/AvailabilityMap';
import CloseButton from './CloseButton';

const contentStyle = { width: 1000, maxWidth: 'none' };
const bodyStyle = { padding: '75px 0 0', background: '#333333' };

export default class AvailabilityButton extends Component {
  static propTypes = {
    productName: PropTypes.string.isRequired,
    productCode: PropTypes.string.isRequired,
    hasNearByStores: PropTypes.bool
  };

  static defaultProps = {
    hasNearByStores: true
  };

  constructor(props) {
    super(props);
    this.state = {
      dialogOpen: false,
      display: ''
    };
  }

  handleOpen = () => {
    this.setState({ dialogOpen: true });
  };

  handleClose = () => {
    this.setState({ dialogOpen: false });
  };

  render() {
    const { productName, productCode, hasNearByStores } = this.props;

    return (
      <div >
        <FlatButton
          backgroundColor={hasNearByStores ? '#67cb33' : '#C8C8C8'}
          label={'verifica disponibilitÃ  in negozi vicini'}
          onClick={this.handleOpen}
          disabled={!hasNearByStores}
          labelStyle={{ color: '#fff', fontSize: '16px' }}
          style={{ height: '70px' }}
        />
        <Dialog
          modal={false}
          open={this.state.dialogOpen}
          contentStyle={contentStyle}
          bodyStyle={bodyStyle}
        >
          <CloseButton handleClick={this.handleClose} />
          <AvailabilityMap productName={productName} productCode={productCode} />
        </Dialog>
      </div>
    );
  }
}
