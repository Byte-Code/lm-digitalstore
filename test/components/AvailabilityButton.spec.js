import React from 'react';
import { fromJS } from 'immutable';
import { shallow } from 'enzyme';
import { FlatButton } from 'material-ui';

import AvailabilityButton from '../../app/components/AvailabilityButton';

describe('AvailabilityButton', () => {
  const productName = 'abc';
  const productCode = '123';

  it('should render properly with a closed dialog', () => {
    const allStoreStock = fromJS([0, 1, 2]);
    const result = shallow(
      <AvailabilityButton
        productCode={productCode}
        productName={productName}
        allStoreStock={allStoreStock}
      />
    );
    expect(result).toMatchSnapshot();
    expect(result.state('dialogOpen')).toBe(false);
  });

  it('should display a dialog when the button is clicked', () => {
    const allStoreStock = fromJS([0, 1, 2]);
    const result = shallow(
      <AvailabilityButton
        productCode={productCode}
        productName={productName}
        allStoreStock={allStoreStock}
      />
    );
    result.find(FlatButton).simulate('click');
    expect(result.state('dialogOpen')).toBe(true);
    expect(result).toMatchSnapshot();
  });
});
