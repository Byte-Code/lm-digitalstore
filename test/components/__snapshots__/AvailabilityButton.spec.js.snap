// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AvailabilityButton should display a dialog when the button is clicked 1`] = `
<div>
  <FlatButton
    backgroundColor="#67cb33"
    disabled={false}
    fullWidth={false}
    label="verifica disponibilità in negozi vicini"
    labelPosition="after"
    labelStyle={
      Object {
        "color": "#fff",
        "fontSize": "16px",
      }
    }
    onClick={[Function]}
    onKeyboardFocus={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    onTouchStart={[Function]}
    primary={false}
    secondary={false}
    style={
      Object {
        "height": "70px",
      }
    }
  />
  <Dialog
    autoDetectWindowHeight={true}
    autoScrollBodyContent={false}
    bodyStyle={
      Object {
        "background": "#333333",
        "padding": "75px 0 0",
      }
    }
    contentStyle={
      Object {
        "maxWidth": "none",
        "width": 1000,
      }
    }
    modal={false}
    open={true}
    repositionOnUpdate={true}
  >
    <CloseButton
      backgroundColor="#333333"
      fill="#858585"
      handleClick={[Function]}
      right={0}
      top={0}
    />
    <withState(withState(withState(withState(withHandlers(Connect(AvailabilityMap))))))
      productCode="123"
      productName="abc"
    />
  </Dialog>
</div>
`;

exports[`AvailabilityButton should render properly with a closed dialog 1`] = `
<div>
  <FlatButton
    backgroundColor="#67cb33"
    disabled={false}
    fullWidth={false}
    label="verifica disponibilità in negozi vicini"
    labelPosition="after"
    labelStyle={
      Object {
        "color": "#fff",
        "fontSize": "16px",
      }
    }
    onClick={[Function]}
    onKeyboardFocus={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    onTouchStart={[Function]}
    primary={false}
    secondary={false}
    style={
      Object {
        "height": "70px",
      }
    }
  />
  <Dialog
    autoDetectWindowHeight={true}
    autoScrollBodyContent={false}
    bodyStyle={
      Object {
        "background": "#333333",
        "padding": "75px 0 0",
      }
    }
    contentStyle={
      Object {
        "maxWidth": "none",
        "width": 1000,
      }
    }
    modal={false}
    open={false}
    repositionOnUpdate={true}
  >
    <CloseButton
      backgroundColor="#333333"
      fill="#858585"
      handleClick={[Function]}
      right={0}
      top={0}
    />
    <withState(withState(withState(withState(withHandlers(Connect(AvailabilityMap))))))
      productCode="123"
      productName="abc"
    />
  </Dialog>
</div>
`;
